engine/access/rest/http/routes/transactions.go

```diff
@@ -52,6 +52,57 @@ func GetTransactionByID(r *common.Request, backend access.API, link commonmodels
 	return response, nil
 }
 
+// GetTransactionsByBlockID gets transactions by requested blockID or height.
+func GetTransactionsByBlockID(r *common.Request, backend access.API, link commonmodels.LinkGenerator) (interface{}, error) {
+	req, err := request.NewGetTransactionsByBlockIDRequest(r)
+	if err != nil {
+		return nil, common.NewBadRequestError(err)
+	}
+
+	blockID := req.BlockID
+	if blockID == flow.ZeroID {
+		block, _, err := backend.GetBlockByHeight(r.Context(), req.BlockHeight)
```

Comment:
I think this should use a `storage.Headers` directly instead of querying the backend.